### DISCUSSION TITLE
DefaultButton with menuProps does not obey style={{width: '100%'}}

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [microsoft/fluentui#24231] [Q&A] [answered]
I want to steratch the split button to take all the containing width on a web part but it keeps the width based on the text of it.
Is there a property in the menuProps to apply  style={{width: '100%'}} ?

Thank you

**Comment by micahgodbolt** [comment, marked as answer]
Due to the complex nature of the split button (two buttons wrapped in a span) we have a specific style target for getting CSS onto the container. The `style` tag you are using, applies styles to the primary button, to target the split button container, use the following:

```
<DefaultButton
        styles={{ splitButtonContainer: { width: '100%' } }}
        text="Standard"
        split
      />
```

**Reply by Ofer-Gal** [reply]
That is a start. it shows something mid way to 100%.  Could the problem be that the row it is in is display: 'flex' ? I am not using Stacks like the demo shows.
![image](https://user-images.githubusercontent.com/5054931/187324660-01ab647c-fa9c-4c76-a8b4-5122e7838e6d.png)
